libs/entitlement/qa/src/javaRestTest/java/org/elasticsearch/entitlement/qa/EntitlementsTestRule.java

```diff
@@ -70,15 +82,16 @@ protected void before() throws Throwable {
                 Files.writeString(testPath.resolve("read_write_file"), "");
             }
         };
-        cluster = ElasticsearchCluster.local()
+        var clusterBuilder = ElasticsearchCluster.local()
             .module("entitled", spec -> buildEntitlements(spec, "org.elasticsearch.entitlement.qa.entitled", ENTITLED_POLICY))
-            .module("entitlement-test-plugin", spec -> setupEntitlements(spec, modular, policyBuilder))
+            .module(ENTITLEMENT_TEST_PLUGIN_NAME, spec -> setupEntitlements(spec, modular, policyBuilder))
             .systemProperty("es.entitlements.enabled", "true")
             .systemProperty("es.entitlements.testdir", () -> testDir.getRoot().getAbsolutePath())
-            .setting("xpack.security.enabled", "false")
-            // Logs in libs/entitlement/qa/build/test-results/javaRestTest/TEST-org.elasticsearch.entitlement.qa.EntitlementsXXX.xml
-            // .setting("logger.org.elasticsearch.entitlement", "DEBUG")
-            .build();
+            .setting("xpack.security.enabled", "false");
+        // Logs in libs/entitlement/qa/build/test-results/javaRestTest/TEST-org.elasticsearch.entitlement.qa.EntitlementsXXX.xml
+        // .setting("logger.org.elasticsearch.entitlement", "DEBUG")
+        systemPropertyBuilder.build((k, v) -> clusterBuilder.systemProperty(k, () -> v.apply(testDir.getRoot().toPath())));
```

Comment:
Could this be simplified by have the test rule take in an optional override? The sysprop we set can always be the same, it's for the test plugin.

Reply:
I did that initially for the test with `load_native_libraries`, but had to introduce this complication to build a policy override for a file entitlement using the tempDir

Reply:
In order to pass the tempDir and use it for building the overridden policy the TestDirectory needs to be built first, so we need to use the lazy `clusterBuilder.systemProperty(k, () -> ...)` setter. 
It is more complicated, I agree.
Alternatives are: not use tempDir, but something like e.g. `relative_to: data` (would we allow that in plugins?) and a specific test method (not an existing one like the `fileList` I used)
Or just not test file entitlements override.

Reply:
I just find this functional interface `void build(BiConsumer<String, Function<Path, String>> adder);` really hard to grok. Instead of the `SystemPropertyBuilder` interface couldn't we simplify this to `Function<Path, SystemPropertyProvider>`? At least then the impl just looks like `path -> { // return a map }` and on the consumer side we just do `cluster.systemProperties(provider.apply(testDir.getRoot().toPath())`.

Reply:
> on the consumer side we just do cluster.systemProperties(provider.apply(testDir.getRoot().toPath())

That wound not work, wouldn't it? That would try to read `testDir` before it's created, resulting in an exception. But I will try to simplify and remove one level of indirection.

Reply:
I had to do the other way around (`Function<Path, Map<String, String>>` but it's simpler now. I also created a named functional interface so it's clearer to read.